Assertion Error of unexpected job status can happen with JOB_QUEUE_DO_KILL_NODE_FAILURE

https://github.com/equinor/ert/blob/b828844409c74bcd22bd57f7196c9861a4bb8694/src/ert/_c_wrappers/job_queue/job_queue_node.py#L249-L252

this has now been observed while debugging a user case. while running esmda, the following exception occurs during the first iteration (`iter-0`) for realization 41:

```
Exception in thread Thread-53:
Traceback (most recent call last):
  File "/opt/rh/rh-python38/root/usr/lib64/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/opt/rh/rh-python38/root/usr/lib64/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/prog/res/komodo/2023.05.02-py38-rhel7/root/lib64/python3.8/site-packages/ert/_c_wrappers/job_queue/job_queue_node.py", line 249, in _job_monitor
    raise AssertionError(
AssertionError: Unexpected job status type after running job: JOB_QUEUE_DO_KILL_NODE_FAILURE
```

ert keeps spinning, moves on to iteration 1, and fails during init with

```
Could not load node: INIT_FILES iens: 41 report: 0
```

(cf. also [internal slack thread](https://equinor.slack.com/archives/C01MCMWD075/p1687501135192689) and further discussion in channel)

cf. also 

- [an old occurence of the error in issue #1566](https://github.com/equinor/ert/issues/1566),
- a [general issue on unhandled job queue states](https://github.com/equinor/ert/issues/1900), and
- [an in-depth comment on the conditions around this error state](https://github.com/equinor/libres/pull/966#issue-646382248)

--- a/src/clib/lib/job_queue/job_node.cpp
+++ b/src/clib/lib/job_queue/job_node.cpp
@@ -67,7 +67,7 @@ struct job_queue_node_struct {
     time_t sim_start;
     /** When did the job finish successfully */
     time_t sim_end;
-    /** Max waiting between sim_start and confirmed_running is 2 minutes */
+    /** Max waiting between sim_start and confirmed_running in seconds*/
     time_t max_confirm_wait;
     /** Timestamp of the status update update file. */
     time_t progress_timestamp;
@@ -281,7 +281,7 @@ job_queue_node_alloc(const char *job_name, const char *run_path,
     node->sim_start = 0;
     node->sim_end = 0;
     node->submit_time = time(NULL);
-    node->max_confirm_wait = 60 * 2; // 2 minutes before we consider job dead.
+    node->max_confirm_wait = 60 * 10;
 
     pthread_mutex_init(&node->data_mutex, NULL);
     return node;
@@ -450,7 +450,7 @@ job_status_type job_queue_node_refresh_status(job_queue_node_type *node,
         double runtime = job_queue_node_time_since_sim_start(node);
         if (runtime >= node->max_confirm_wait) {
             logger->info("max_confirm_wait ({}) has passed since sim_start"
-                         "without success; {} is dead (attempt {})",
+                         "without success; {} is assumed dead (attempt {})",
                          node->max_confirm_wait, node->job_name,
                          node->submit_attempt);
             job_status_type new_status = JOB_QUEUE_DO_KILL_NODE_FAILURE;

--- a/src/ert/job_queue/job_queue_node.py
+++ b/src/ert/job_queue/job_queue_node.py
@@ -276,7 +276,10 @@ class JobQueueNode(BaseCClass):  # type: ignore
                     )
                     self._set_status(JobStatusType.JOB_QUEUE_FAILED)
                     self.run_exit_callback()
-            elif current_status == JobStatusType.JOB_QUEUE_IS_KILLED:
+            elif current_status in [
+                JobStatusType.JOB_QUEUE_IS_KILLED,
+                JobStatusType.JOB_QUEUE_DO_KILL_NODE_FAILURE,
+            ]:
                 self.run_exit_callback()
             elif current_status == JobStatusType.JOB_QUEUE_FAILED:
                 logger.error(

